src/Plugin.php

```diff
@@ -80,7 +85,7 @@ public static function wp_update_nav_menu(){
    */
   public static function wp_get_nav_menu_items($items, $menu, $args) {
     foreach($items as $key => $item) {
-      if (get_post_meta($item->ID, '_menu_item_status', TRUE) === 'checked') {
+      if ((bool) get_post_meta($item->ID, '_menu_item_status', TRUE) === FALSE) {
```

Comment:
The cast is applied to the whole expression; i.e., the result of the expression `get_post_meta() === FALSE` (which is already a Boolean) is casted into a Boolean.

In order to cast the value before the expression is evaluated, you either need to add parenthesis, or you need to switch the sides of the comparison, because (most) **PHP code is evaluated right-to-left** (which I usually prefer):

```diff
-if ((bool) get_post_meta($item->ID, '_menu_item_status', TRUE) === FALSE) {
+if (FALSE === (bool) get_post_meta($item->ID, '_menu_item_status', TRUE)) {
```
Stating the comparison value first is also known as [Yoda condition](https://en.wikipedia.org/wiki/Yoda_conditions).  We only use them when they technically make sense; i.e., in situations like this.

However, your new comparison will hide all menu items for which no value exists; e.g., after installing the plugin (because `NULL` is cast to `FALSE`).  I don't think we want to force users to edit all of their menus after installing the plugin, just to mark all menu items as "visible"…?

Reply:
👌 about the Yoda condition, thanks!

About the default status of hidden value… I deeply tested it. 
As soon you install the plugin, the value returned by `(bool) get_post_meta($item->ID, '_menu_item_status', TRUE)` is `TRUE` so no link is hidden at all. If you mark it as `Hidden` into the backend, then it turns to `FALSE` and it hides the link.